pkg/openservicebroker/validation.go

```diff
@@ -0,0 +1,47 @@
+package openservicebroker
+
+import (
+	"fmt"
+	"regexp"
+
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	"github.com/openshift/origin/pkg/template/api/validation"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+func ValidateProvisionRequest(preq *ProvisionRequest) field.ErrorList {
+	errors := ValidateUUID(field.NewPath("service_id"), preq.ServiceID)
+	errors = append(errors, ValidateUUID(field.NewPath("plan_id"), preq.PlanID)...)
+
+	for key := range preq.Parameters {
+		if !validation.ParameterNameRegexp.MatchString(key) &&
+			key != templateapi.NamespaceParameterKey &&
+			key != templateapi.ImpersonateUserParameterKey {
+			errors = append(errors, field.Invalid(field.NewPath("parameters."+key), key, fmt.Sprintf("does not match %v", validation.ParameterNameRegexp)))
+		}
+	}
```

Comment:
This is already done in the next layer down - I was trying to keep pkg/openservicebroker more generic.